contracts/core/OriginCore.sol

```diff
@@ -285,7 +302,52 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
         external
         returns (bool success_)
     {
-        revert("Method not implemented.");
+        /**
+         * This check if valid meta-block proposal exists for given
+         * kernel hash and transition hash. This will revert transaction
+         * if proposal doesn't exist.
+         */
+        checkMetaBlockProposal(
+            _kernelHash,
+            _transition
+        );
+
+        bytes32 voteHash = MetaBlock.hashVote(
+            _coreIdentifier,
+            _transition,
+            _source,
+            _target,
+            _sourceHeight,
+            _targetHeight
+        );
+
+        uint256 weight;
+        address signer;
+
+        /**
+         * This check the validity of vote, on successful validation it returns
+         * address of validator and weight at current meta-block.
+         * This will revert transaction if vote validation fails.
+         */
+        (signer, weight) = checkVoteValidity(
+            voteHash,
+            _v,
+            _r,
+            _s
+        );
+        /** This saves vote in the seal. */
```

Comment:
```suggestion
        /* This saves vote in the seal. */
```
I would even argue that you don't need this comment, as the function's name is `saveVote`, which makes it clear to the reader that the vote is saved here.